rmw_fastrtps_dynamic_cpp/include/rmw_fastrtps_dynamic_cpp/MessageTypeSupport_impl.hpp

```diff
@@ -34,9 +35,14 @@ MessageTypeSupport<MembersType>::MessageTypeSupport(const MembersType * members)
   assert(members);
   this->members_ = members;
 
-  std::string name = std::string(members->package_name_) + "::msg::dds_::" +
-    members->message_name_ + "_";
-  this->setName(name.c_str());
+  std::ostringstream ss;
+  std::string message_namespace(this->members_->message_namespace_);
+  std::string message_name(this->members_->message_name_);
+  if (!message_namespace.empty()) {
+    ss << message_namespace << "::";
+  }
+  ss << "dds_::" << message_name << "_";
```

Comment:
Couldn't this call `_create_type_name()` instead of duplicating the logic?

Same below.

Reply:
It is declared the `src` directory. I'm not sure if there's a valid reason to keep the header in the `src` directory or if we can move it to `include` so it is reachable by this header.